app/models/data.py

```diff
@@ -96,26 +135,35 @@ class CoreData(db.Model, DataMixin):
     # What other columns are we missing?
     sourceNotes = db.Column(db.String)
 
-    def __init__(self, **kwargs):
-        super(CoreData, self).__init__(**kwargs)
-
     @hybrid_property
     def lastUpdateEt(self):
         # convert lastUpdateTime (UTC) to ET
-        raise NotImplementedError
+        return self.lastUpdateTime.replace(tzinfo=pytz.utc).astimezone(pytz.timezone('US/Eastern'))
 
     @hybrid_property
     def checkTimeEt(self):
         # convert lastCheckTime (UTC) to ET
-        raise NotImplementedError
+        return self.lastCheckTime.replace(tzinfo=pytz.utc).astimezone(pytz.timezone('US/Eastern'))
 
     @hybrid_property
     def totalTestResults(self):
         # Calculated value (positive + negative) of total test results.
-        raise NotImplementedError
+        if self.positive is None or self.negative is None:
+            return None
+        return self.positive + self.negative
 
-    # TODO: make a recursive to_dict in the base class to include relationships
-    def to_dict(self):
-        d = super(CoreData, self).to_dict()
-        d['batch'] = self.batch.to_dict()
-        return d
+    def __init__(self, **kwargs):
+        # strip any empty string fields from kwargs
+        kwargs = {k: v for k, v in kwargs.items() if v is not None and v != ""}
+
+        # convert lastUpdateIsoUtc to lastUpdateTime
+        kwargs['lastUpdateTime'] = parser.parse(kwargs['lastUpdateIsoUtc'])
+
+        # FOR NOW defaulting lastCheckTime to lastUpdateTime; NEED TO FIX
+        # ALSO FOR NOW defaulting "date" to today
+        kwargs['lastCheckTime'] = kwargs['lastUpdateTime']
+        kwargs['date'] = date.today()
```

Comment:
do we want to call this one `dataDate`? I think we decided that's the least bad answer for this poor field. 

do we/should we have a key in place that ensures (state, batchId, date) is unique in coreData, so there can't be any ambiguity if there are somehow duplicates in a batch?